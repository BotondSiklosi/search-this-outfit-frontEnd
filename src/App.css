* {
    margin: 0;
    padding: 0;
}

html {
    font-size: 10px;
    font-family: "Lato", Arial, sans-serif;
}

body {
    height: 100vh;
    width: 100%;
    color: #fff;
    background: linear-gradient(-45deg, #EE7752, #b62c61, #23A6D5, #23bad5);
    background-size: 400% 400%;
    position: relative;
    animation: change 30s ease-in-out infinite;
}

@keyframes change {
    0% {
        background-position: 0 50%;
    }
    50% {
        background-position: 100% 50%;
    }
    100% {
        background-position: 0 50%;
    }
}

.App {
    padding-right: 15px;
    padding-left: 15px;
    margin-right: auto;
    margin-left: auto;
}

.main-page h4 {
    font-size: 5rem;
    text-transform: uppercase;
    letter-spacing: 2px;
    position: relative;

}

.main-page {
    display: grid;
    justify-content: center;
}


.custom-file-btn {
    background: #68d7cca3;
    color: #fff;
    width: 10vw;
    height: 3vh;
    border: 1px solid #338033;
    border-radius: 6px;
    overflow: hidden;
}