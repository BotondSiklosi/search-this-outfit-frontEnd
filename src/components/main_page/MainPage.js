import React, {useContext} from "react";
import InputField from "./InputFile";
import { PictureContext} from "../context/PictureContext";
import SideBar from "./SideBar";
import PictureResults from "./PictureResults";

function MainPage() {

    const {
        pictureURL
    } = useContext(PictureContext)

    return (
        <div className="main-page" >
            <div className="image-upload-container">
                <InputField />
                {pictureURL !== null && <img className="current-image" src={pictureURL} alt=""/>}
            </div>

            <SideBar />
            <PictureResults />
        </div>
    )
}

export default MainPage