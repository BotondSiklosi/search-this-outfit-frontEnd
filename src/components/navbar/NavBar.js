import React from "react";
import styled from "styled-components";


const NavBarStyle = styled.div`
  .nav {
    background: rgb(111, 110, 110);
    background: linear-gradient(180deg, rgb(172, 167, 167) 0%, rgba(220, 217, 217, 0.35) 100%);
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    display: flex;
    height: 8vh;
    align-items: center;
    z-index: 100;
    justify-content: space-between;
}



.nav a {
    text-decoration: none;
    color: white;

}

.title-a {
    position: relative;
    left: 5%;
}

.nav h4 {
    flex-wrap: wrap;
    display: flex;
    justify-content: center;
    align-content: center;
    font-size: 1.5rem;
    text-transform: uppercase;
    letter-spacing: 2px;
    width: 120px;
}

@media screen and (max-height: 650px) {
    .nav h4 {
        font-size: 1rem;
    }
}

.nav-links {
    display: grid;
    grid-template-columns: repeat(4, 1fr);
    width: 35%;
}

.nav-links a {
    display: flex;
    justify-content: center;
    align-content: center;
    font-weight: bold;
    font-size: 1.5rem;
    margin-right: 5rem;
}

.nav-links a:hover {
    background-color: rgba(220, 217, 217, 0.35);
}

.burger {
    position: absolute;
    right: 6%;
    display: none;
}

.burger div {
    width: 25px;
    height: 5px;
    margin: 5px;
    background-color: white;
}


@media screen and (max-width: 1024px) {
    body {
        overflow-x: hidden;
    }

    .nav-links {
        position: fixed;
        right: -250px;
        height: 92vh;
        top: 8vh;
        background-color: rgba(220, 217, 217, 0.5);
        display: flex;
        flex-direction: column;
        align-items: center;
        justify-content: space-around;
        padding-bottom: 50vh;
        transition: right 0.5s ease-in-out, opacity 0.4s ease 0.2s;
        width: 250px;
        opacity: 0;
    }

    .burger {
        display: block;
        cursor: pointer;
    }
}

.nav-active {
    opacity: 1;
    right: 0;
}

`;


function NavBar() {


    const navSlide = () => {
        const nav = document.querySelector('.nav-links');
        nav.classList.toggle('nav-active')

    }

    return (
        <NavBarStyle>
            <div className="nav" data-testid="navbar">
                <a className="title-a" href="/">
                    <h4 className="title">

                        Search <i className="fas fa-user-tie"/>This Outfit
                    </h4>
                </a>
                <div className="nav-links">
                    <a href="/">link1</a>
                    <a href="/">link2</a>
                    <a href="/">link3</a>
                    <a href="/">link4</a>
                </div>
                <div className="burger" onClick={navSlide}>
                    <div className="line-1"></div>
                    <div className="line-2"></div>
                    <div className="line-3"></div>
                </div>
            </div>
        </NavBarStyle>
    )
}

export default NavBar