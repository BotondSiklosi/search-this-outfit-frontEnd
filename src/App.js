import React from 'react';
import './App.css';
import {BrowserRouter as Router, Route} from "react-router-dom"
import MainPage from "./components/main_page/MainPage";
import {PictureProvider} from "./context/PictureContext";
import NavBar from "./components/navbar/NavBar";
import GlobalStyle from "./components/style-components/styled/GlobalStyle";
import Home from "./components/home/Home";
import About from "./components/about/About";
import Contacts from "./components/contacts/Contacts";



function App() {
    return (
        <div className="App" data-testid="app">

            <PictureProvider>
                <GlobalStyle />

                    <div className="backGround" />
                    <Router>
                        <NavBar/>
                        <div className="container">
                            <Route exact path="/" component={Home}/>
                            <Route exact path="/search" component={MainPage}/>
                            <Route exact path="/about" component={About}/>
                            <Route exact path="/contacts" component={Contacts}/>
                        </div>
                    </Router>
            </PictureProvider>
        </div>
    );
}

export default App;
