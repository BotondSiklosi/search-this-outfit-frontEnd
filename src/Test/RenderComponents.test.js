import React from 'react';
import App from '../App';
import NavBar from "../components/navbar/NavBar";
import MainPage from "../components/main_page/MainPage";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

describe('App.js', () => {
    it('Render App correctly', () => {
        const component = shallow(<App />);

        expect(component).toMatchSnapshot();
    });
});

describe('NavBar.js', () => {
    it('Renders NavBar correctly', () => {
        const component = shallow(<NavBar/>);

        expect(component).toMatchSnapshot();
    });
});


describe('MainPage.js', () => {
    it('Renders MainPage correctly', () => {
        const component = shallow(<MainPage />);

        expect(component).toMatchSnapshot();
    });
});