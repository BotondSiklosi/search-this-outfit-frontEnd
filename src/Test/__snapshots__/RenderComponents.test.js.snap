// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.js Render App correctly 1`] = `ShallowWrapper {}`;

exports[`NavBar.js Renders NavBar correctly 1`] = `ShallowWrapper {}`;
